typeDefs.js

```diff
@@ -231,6 +231,7 @@ const typeDefs = gql`
 
     getCommentById(id: ID!): CommentResult
     getAllCommentsByProduct(productId: ID!): [CommentResult]
+    getAllCommentsByUser(userEmail: String!): [CommentResult]
```

Comment:
Why do you return `CommentResult` if no error can be returned from the call of this query? Also `CommentResult` should not be an array item because we return an error or the data, but not the mixed array where errors and valid data items are both presented